sos/upload/targets/redhat.py

```diff
@@ -100,9 +81,6 @@ def get_upload_url(self):
                 return self.commons['cmdlineopts'].upload_url
             if self.commons['cmdlineopts'].upload_protocol == 'sftp':
                 return self.RH_SFTP_HOST
-            if not self.commons['cmdlineopts'].case_id and not\
-               self.prompt_for_case_id():
-                return self.RH_SFTP_HOST
```

Comment:
I am presuming this was removed on purpose and you don't need this if a case id is not there?

Reply:
Also not sure why this is being removed.

Reply:
Indeed, quite basic use case `sos report -o host --batch --upload` diverts: it tries to upload to the customer portal while it should use SFTP (no case id).

Instead of 3 lines removal, there must be a change on 2nd line to:

    self.commons['policy'].prompt_for_case_id(self.commons['cmdlineopts']):

I had to overview this removal, sorry for ACKing it.

Reply:
The reason why I removed it was to run some tests and then I pushed the commit without re-adding it, sorry. 
Thank you @arif-ali ++ for spotting this :)
I've just pushed a new commit with the code back